src/status/status.factory.js

```diff
@@ -0,0 +1,209 @@
+const { stripIndent } = require('common-tags');
+
+
+module.exports = (logProvider, stateProvider, logger) => {
+    //Helper functions
+    function activeMembers(events) {
+        for(var i=0; i < event.length; i++){
+            var obj = events.logs[i];
+                if(obj.action == "Participated"){
+                    return ({obj})
+                }
+            }
+    }
+
+    function activeEvents(events){
+     for(var i=0; i < events.team_id.length; i++){
+         var obj = events;
+         if(obj.halted == true){
+                return ({obj});
+            }
+        }
+    }
+
+    function happenedInPeriodUser(date, action, back){
+        const alerts = ({
```

Comment:
you don't need to create an object here to override it later completely. Move the creation to the end of each branch of the conditional statement